src/contexts/Auth/AuthProvider.jsx

```diff
@@ -0,0 +1,214 @@
+import React, { Component } from 'react'
+import auth0 from 'auth0-js'
+import { withRouter } from 'react-router'
+
+import { ConfigurationContext } from '../Configuration'
+import { alert } from 'services'
+
+import { authUser } from './queries'
+
+export const AuthContext = React.createContext()
+
+class AuthProvider extends Component {
+  static contextType = ConfigurationContext
+
+  constructor(props) {
+    super(props)
+
+    const { session, user } = this.retrieveFromLS()
+
+    this.state = {
+      session,
+      user,
+      actions: {
+        login: this.login,
+        logout: this.logout,
+        parseHash: this.parseHash,
+        renewAuth: this.renewAuth
+      }
+    }
+  }
+
+  componentDidMount() {
+    const { session, user } = this.state
+    if (session && user) this.scheduleRenew()
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const conf = this.context
+
+    if (!conf.loading && !prevState.auth0 && !this.state.auth0) {
+      this.setState({ auth0: this.initAuth0(conf), ready: true })
+    }
+  }
+
+  render() {
+    return <AuthContext.Provider value={this.state}>{this.props.children}</AuthContext.Provider>
+  }
+
+  /* Auth lifecyle */
+
+  login = redirectTo => {
```

Comment:
This file mixes React stuff with non-React stuff, while the old auth service was pure non-React. I think it would be better if the non-React remained in its own file.

Reply:
I wasn't satisfied with how the old auth service worked, but I indeed coupled the new AuthProvider with React stuff.

I just commited a new auth service which is non-react stuff only and is (I think) better than the old one :)